### DISCUSSION TITLE
Coloring Lines on PolyData

### DISCUSSION THREAD

**Discussion by kjans123** [pyvista/pyvista#2277] [Q&A] [answered]
I have the below set of lines:
![image](https://user-images.githubusercontent.com/35381053/155767234-88a31568-d729-4742-9026-d94261f4449a.png)
that I generate with 
```
mesh = pv.PolyData(vertices, lines=np.hstack(coupled_lines))
```

Is there any way to color each line in a PolyData object? I can do this with a Plotter object, but the exporting outputs are pretty limited with that instance.

**Comment by banesullivan** [comment, marked as answer]
You need to add point or cell data to your `PolyData` mesh. Reference https://docs.pyvista.org/user-guide/data_model.html#data-arrays

```py
mesh = pv.PolyData(vertices, lines=np.hstack(coupled_lines))
mesh['my data'] = np.random.rand(mesh.n_points)
```
